src/active-response/netsh.c

```diff
@@ -12,14 +12,60 @@
 #include "active_responses.h"
 
 #define RULE_NAME "WAZUH ACTIVE RESPONSE BLOCKED IP"
-#define NETSH "C:\\Windows\\System32\\netsh.exe"
+#define NETSH     "C:\\Windows\\System32\\netsh.exe"
+#define REG       "C:\\Windows\\System32\\reg.exe"
+
+#define PATH_FIREWALL_PROFILES_REG_DEFAULT "HKEY_LOCAL_MACHINE\\SYSTEM\\CurrentControlSet\\Services\\SharedAccess\\Parameters\\FirewallPolicy\\"
+#define FIREWALL_DATA_INITIALIZE {false, false, FIREWALL_DOMAIN}
+#define FIREWALL_PROFILES_MAX (3)   /*!< Maximun number of profiles*/
+
+typedef struct {
```

Comment:
I think this struct is not necessary. You can create `log_msg` locally inside the function `getAllProfilesStatus`. Only send argv[0] as parameter of `getAllProfilesStatus`.